lib/string.js

```diff
@@ -91,19 +91,34 @@ internals.String = class extends Any {
         });
     }
 
-    regex(pattern, name) {
+    regex(pattern, patternOptions) {
 
         Hoek.assert(pattern instanceof RegExp, 'pattern must be a RegExp');
 
-        pattern = new RegExp(pattern.source, pattern.ignoreCase ? 'i' : undefined);         // Future version should break this and forbid unsupported regex flags
+        const patternObject = {
+            pattern: new RegExp(pattern.source, pattern.ignoreCase ? 'i' : undefined)         // Future version should break this and forbid unsupported regex flags
+        };
+
+        const patternIsInverted = (typeof patternOptions === 'object' && patternOptions.invert);
```

Comment:
For `patternObject.name = patternOptions.name;` it looks like I need to have either a ternary check or a fallback value for `describe` when name isn't provided:
```
 "name": "[undefined]"
```

Preference here?